dbt/include/glue/macros/materializations/incremental/incremental.sql

```diff
@@ -57,17 +47,29 @@
         {% if file_format == 'delta' %}
             {{ adapter.delta_create_table(target_relation, sql, unique_key, partition_by, custom_location) }}
             {% set build_sql = "select * from " + target_relation.schema + "." + target_relation.identifier + " limit 1 " %}
+        {% elif file_format == 'iceberg' %}
+            {{ adapter.iceberg_write(target_relation, sql, unique_key, partition_by, custom_location, strategy, table_properties) }}
+            {% set build_sql = "select * from glue_catalog." + target_relation.schema + "." + target_relation.identifier + " limit 1 "%}
         {% else %}
             {% set build_sql = create_table_as(False, target_relation, sql) %}
         {% endif %}
-      {% elif existing_relation_type == 'view' or full_refresh_mode %}
-        {{ drop_relation(target_relation) }}
+      {% elif existing_relation_type == 'view' or should_full_refresh() %}
+        {{ glue__drop_relation(target_relation) }}
```

Comment:
nit: drop_relation should be enough. `glue__drop_relation` is automatically called when you call `drop_relation` as you are in the context of an adapter. See what we do here https://github.com/dbt-athena/dbt-athena/blob/main/dbt/include/athena/macros/materializations/models/table/table.sql#L66

Reply:
thanks